sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,380 @@
+                for field in self.statedb_json_sync_fields:
+                    json_file_dict[device][field] = self.state_db.hget('STORAGE_INFO|{}'.format(device), field)
+            
+            self.fsio_sync_time = time.time()
+            json_file_dict["successful_sync_time"] = str(self.fsio_sync_time)
+
+            with open(FSIO_RW_JSON_FILE, 'w+') as f:
+                json.dump(json_file_dict, f)
+                self.state_db.hset("{}|{}".format(STORAGE_DEVICE_TABLE,FSSTATS_SYNC_TIME_KEY), "successful_sync_time", str(self.fsio_sync_time))
+
+        except Exception as ex:
+            self.log_error("Unable to sync state_db to disk: {}".format(str(ex)))
+
+    # Run a sanity check on the state_db. If successful, get total, latest 
+    # FSIO reads and writes for each storage device from STATE_DB
+    def _load_fsio_rw_statedb(self):
+
+        # Sanity Check:
+        
+        # If the number of STORAGE_INFO|* keys does not equal the 
+        # number of storage disks on the device + FSSTATS_SYNC field,
+        # there has been a corruption to the database. In this case we 
+        # pivot to the JSON file being the Source of Truth.
+        try:
+            if (len(self.state_db.keys("STORAGE_INFO|*")) != (len(self.storage.devices) + 1)):
+                return
+
+            # For each storage device on the switch,
+            for storage_device in self.storage.devices:
+
+                # Get the total and latest procfs reads and writes from STATE_DB
+                for field in self.statedb_json_sync_fields:
+                    value = self.state_db.hget('STORAGE_INFO|{}'.format(storage_device), field)
+                    self.fsio_rw_statedb[storage_device][field] = "0" if value is None else value
+
+                    if value is None: self.log_warning("{}:{} value = None in StateDB".format(storage_device, field))
+            
+            self.statedb_storage_info_loaded = True
+        except Exception as e:
+            self.log_error("Reading STATE_DB failed with: {}".format(str(e)))
+
+
+    def _determine_sot(self):
+
+        # This daemon considers the storage information values held in the STATE_DB to be its
+        # Source of Truth. 
+
+        # If the state_db information did not load successfully but the JSON file did, 
+        # we consider the JSON file to be the SoT.
+
+        if self.statedb_storage_info_loaded == False and self.fsio_json_file_loaded == True:
+            self.use_fsio_json_baseline = True
+            self.use_statedb_baseline = False
+
+        # If stormond is coming back up after a daemon crash, storage information would be saved in the 
+        # STATE_DB. In that scenario, we use the STATE_DB information as the SoT and reconcile the FSIO
+        # reads and writes values.
+        elif self.statedb_storage_info_loaded == True:
+                self.use_fsio_json_baseline = False
+                self.use_statedb_baseline = True
```

Comment:
@assrinivasan can you make the logic more clear, i.e, if the stats are available in STATE_DB, then use that and as a fallback use .json values from the backup

Reply:
Done in latest